test/math.jl

```diff
@@ -1498,6 +1498,15 @@ end
     n = Int64(1024 / log2(E))
     @test E^n == Inf
     @test E^float(n) == Inf
+
+    # #55633
+    struct Issue55633 <: Number end
+    struct Issue55633Squared <: Number end
+    Base.:(*)(::Issue55633, ::Issue55633) = Issue55633Squared()
+    Base.promote_rule(::Type{Issue55633}, ::Type{Issue55633Squared}) = Int
+    Base.convert(::Type{Int}, ::Issue55633) = 3
+    Base.convert(::Type{Int}, ::Issue55633Squared) = 9
+    @test 9 === Issue55633()^2
 end
```

Comment:
I suspect some of these tests will fail:

```suggestion
    for (x,x0,x1,x2) in ((im, 1+0im, 0+1im, -1+0im),
                         ('x', "", "x", "xx"),
                         (pi, 1.0, Float64(pi), pi*pi))
        @test x^0 === x0
        @test x^1 === x1
        @test x^2 === x2
    end
end
```

Reply:
I don't think strings are supposed to be supported? The code relies on `copy`, but there's no `copy(::String)`.

Reply:
Ah, sure enough, there's a different `^` for those — `power_by_squaring('x', 0)` doesn't even get that far because it tries to convert the char to a string first.  Maybe use Furlongs instead?  Just trying to think of all the odd types here.

Reply:
`^` is broken for Furlongs both before and after this change. And a furlong squared isn't promotable together with the original furlong. This is really issue #24151. I don't think there's a good way of fixing that issue.

I propose we just keep `^` broken for furlongs.